src/hooks/usePartnerTheme.ts

```diff
@@ -0,0 +1,112 @@
+'use client';
+
+import { useTheme } from '@mui/material';
+import { useMemo } from 'react';
+import { STRAPI_PARTNER_THEMES } from 'src/const/strapiContentKeys';
+import { useSettingsStore } from 'src/stores/settings';
+import type { PartnerThemesData } from 'src/types/strapi';
+import { shallow } from 'zustand/shallow';
+import { useIsHomepage } from './useIsHomepage';
+import { useStrapi } from './useStrapi';
+
+export const usePartnerTheme = () => {
+  const theme = useTheme();
+  const isHomepage = useIsHomepage();
+  const setThemeMode = useSettingsStore((state) => state.setThemeMode);
+  const partnerThemeUid = useSettingsStore(
+    (state) => state.partnerThemeUid,
+    shallow,
+  );
+  const {
+    data: partnerThemes,
+    isSuccess,
+    url,
+  } = useStrapi<PartnerThemesData>({
+    contentType: STRAPI_PARTNER_THEMES,
+    queryKey: ['partner-themes-filter', partnerThemeUid],
+    filterUid: partnerThemeUid,
+  });
+
+  const imageUrl = useMemo(() => {
+    if (
+      partnerThemeUid &&
+      partnerThemes?.length > 0 &&
+      isSuccess &&
+      (partnerThemes[0].attributes.BackgroundImageLight.data?.attributes.url ||
+        partnerThemes[0].attributes.BackgroundImageDark.data?.attributes.url)
+    ) {
+      return theme.palette.mode === 'light'
+        ? new URL(
+            partnerThemes[0].attributes.BackgroundImageLight.data?.attributes.url,
+            url.origin,
+          )
+        : new URL(
+            partnerThemes[0].attributes.BackgroundImageDark.data?.attributes.url,
+            url.origin,
+          );
+    } else {
+      return undefined;
+    }
+  }, [
+    partnerThemeUid,
+    partnerThemes,
+    isSuccess,
+    theme.palette.mode,
+    url.origin,
+  ]);
+
+  const backgroundColor = useMemo(() => {
+    if (partnerThemeUid && partnerThemes?.length > 0 && isSuccess) {
+      return theme.palette.mode === 'light'
+        ? partnerThemes[0].attributes.LightBackgroundColor
+        : partnerThemes[0].attributes.DarkBackgroundColor;
+    } else {
+      return undefined;
+    }
+  }, [isSuccess, partnerThemeUid, partnerThemes, theme.palette.mode]);
+
+  const availableWidgetTheme = useMemo(() => {
+    if (partnerThemeUid && partnerThemes?.length > 0 && isSuccess) {
+      if (
+        partnerThemes[0].attributes.lightConfig &&
+        partnerThemes[0].attributes.darkConfig
+      ) {
+        return 'system';
+      } else if (partnerThemes[0].attributes.darkConfig) {
+        setThemeMode('dark');
+        return 'dark';
+      } else {
+        setThemeMode('light');
+        return 'light';
+      }
+    }
+  }, [isSuccess, partnerThemeUid, partnerThemes, setThemeMode]);
+
+  const currentWidgetTheme = useMemo(() => {
+    if (availableWidgetTheme === 'system') {
+      return theme.palette.mode === 'light'
+        ? partnerThemes[0].attributes.lightConfig
```

Comment:
Way too many dark|light mode check, isn't it possible to refactorize it instead of doing the check 6 times into this file? :(